### ISSUE TITLE
stability issues when uploading long activities

### ISSUE THREAD

**Issue by dhiltonp** [jonasoreland/runnerup#277]
Somewhere between 8 and 10 hours, uploading becomes problematic.

Runkeeper works
RunningAhead works if selected by itself
Endomondo works if selected by itself

**Comment by dhiltonp** [comment]
Ugh... Mobile

Strava and Garmin do not work at all.

I'm currently on a bike tour and am limited to a phone.

**Comment by dhiltonp** [comment]
I could probably send a db dump of it would help?

**Comment by jonasoreland** [comment]
my guess is that it's out of memory...
and a db dump sounds good.

my guess would be a solution where we don't upload every gps location,
but every third or so...depending on how long the acvtivity is...

/Jonas

**Comment by skaldarnar** [comment]
Either that or try to come up with a good interpolation to throw out some points (ideally the "not so important" ones). I'm planning to have a look at the graph views, so I might put some thought into this as well....

**Comment by dhiltonp** [comment]
Both of my 10 hour activities included some sections where I forgot to
pause, so auto-collapsing those sections(or me enabling auto-pause) might
help.

I generated tcx files and they were about 8MB, so perhaps the uploaded
could be made to be more memory-efficient?

**Comment by dhiltonp** [comment]
https://www.dropbox.com/s/xia3x7sesjjtbqg/runnerup.db.export?dl=0

**Comment by bagage** [comment]
By the way if you are using default settings (eg GPS update distance frequency: 0 meter), pull request #291 should reduce drastically size of the track and save a lot of battery as well. 
If you are running into unlimited GPS refresh rate, you can end up with pretty big files!

**Comment by jonasoreland** [comment]
why not instead simply change the value in the preferences ?

/Jonas

**Comment by bagage** [comment]
Yes you can do it but there is a bug when using default parameters values which could explain the issue.

**Comment by gerhardol** [comment]
This is a reason for supporting .fit in #605
